server/src/test/java/org/apache/calcite/avatica/remote/RemoteMetaTest.java

```diff
@@ -481,6 +483,38 @@ private static String longString(String fragment, int length) {
     }
   }
 
+  @Test public void testBigDecimalTest() throws Exception {
+    final String tableName = "testbigdecimal";
+    try (Connection conn = DriverManager.getConnection(url);
+          Statement stmt = conn.createStatement()) {
+      conn.setAutoCommit(false);
+      stmt.execute("DROP TABLE IF EXISTS " + tableName);
+      stmt.execute("CREATE TABLE " + tableName + " ("
+          + "pk VARCHAR NOT NULL PRIMARY KEY, "
+          + "v1 DECIMAL(10,5))");
+      conn.commit();
+      try (PreparedStatement pstmt = conn.prepareStatement("INSERT INTO "
+          + tableName + " values(?, ?)")) {
+        pstmt.setString(1, "1");
+        pstmt.setBigDecimal(2, new BigDecimal("12345.67890"));
+        assertEquals(1, pstmt.executeUpdate());
+
+        pstmt.setString(1, "2");
+        pstmt.setObject(2, new BigDecimal("12345.67890"));
+        assertEquals(1, pstmt.executeUpdate());
+
+        pstmt.setString(1, "3");
+        pstmt.setObject(2, new BigDecimal("12345.67890"), Types.NUMERIC);
+        assertEquals(1, pstmt.executeUpdate());
+
+        pstmt.setString(1, "4");
+        pstmt.setObject(2, new BigDecimal("12345.67890"), Types.DECIMAL);
+        assertEquals(1, pstmt.executeUpdate());
+      }
+      conn.commit();
```

Comment:
Wouldn't be a bad idea to validate that we can pull the value back out as a BigDecimal too (if I messed up BigDecimal on the write path, wouldn't be surprised if I also did on the read path :P).

Reply:
Sure, I'll add that tomorrow.

Reply:
Added the tests